tests/performance/apdex-score/settings-provider.js

```diff
@@ -0,0 +1,81 @@
+const { resolve, extname } = require('node:path');
+const { existsSync, readFileSync } = require('node:fs');
+
+let SETTINGS_CACHE;
+
+const getInstanceURL = () => {
+  return SETTINGS_CACHE?.instanceURL;
+};
+
+const getIterations = () => {
+  return SETTINGS_CACHE?.iterations;
+};
+
+const getTestsToSkip = () => {
+  return SETTINGS_CACHE?.skip;
+};
+
+const hasPrivacyPolicy = () => {
+  return !!SETTINGS_CACHE?.hasPrivacyPolicy;
+};
+
+const getCapabilitiesSettings = () => {
+  return SETTINGS_CACHE?.capabilities;
+};
+
+const getUser = (type, role) => {
+  return SETTINGS_CACHE?.users?.find(user => user.type === type && user.role === role);
+};
+
+const getPage = (name) => {
+  return SETTINGS_CACHE?.pages && SETTINGS_CACHE.pages[name];
+};
+
+const getForm = (name) => {
+  return SETTINGS_CACHE?.forms && SETTINGS_CACHE.forms[name];
+};
+
+const getCommonElements = () => {
+  return SETTINGS_CACHE?.commonElements;
+};
+
+const settingsProvider = {
+  getIterations,
+  getInstanceURL,
+  getTestsToSkip,
+  hasPrivacyPolicy,
+  getCapabilitiesSettings,
+  getUser,
+  getPage,
+  getForm,
+  getCommonElements,
+};
+
+module.exports = () => {
+  if (SETTINGS_CACHE) {
+    return settingsProvider;
+  }
```

Comment:
If you create the `loadSettings()` method as suggested, you could make this easier to understand like this:
```
if (!SETTINGS_CACHE) {
    loadSettings();
}
```